src/ddimm_parser.cpp

```diff
@@ -252,6 +252,12 @@ types::VPDMapVariant DdimmVpdParser::parse()
 {
     try
     {
+        if ((constants::DDIMM_11S_BARCODE_START +
```

Comment:
Is this check important for this parser to work?
Or let me re-phrase: IS this check mandatory for this parser to work? Can this parser be used for any other thing even if this check fails?

Reply:
Yes. This is important, as BAR code for DDIMM starts at DDIMM_11S_BARCODE_START index and total length is 26.
This includes "11S", part number(7 bytes), serial number(12 bytes), CCIN(4)
which we are interested in.
we this size is needed for parsing.

Reply:
Then why this class should even get instantiated without this check?

Reply:
Moved the changes to constructor